### PR TITLE
Linear and logistic regression

### LINKED ISSUE #10: Logistic/Linear Regression

### PR DESCRIPTION
**Pull request by ZamiraKholmatova**
Linear and logistic regression

### CODE REVIEW

**Review by artemkurylev — CHANGES_REQUESTED** [review]
Pros:
Structure
Conclusion is good
Nice that you have the visualization of your predictions
Advice:
More charts, for example, showing the distribution of target variables
Maybe some formulas for example about loss functions would be nice
Also, I think it is not a very good idea to leave collab open for editing (this is the most important)

**Review by zakariabeni — CHANGES_REQUESTED** [review]
Hi, 
Overall, it is good and well-structured.
However, I have some suggestions.
- in using @interact for displaying data, I think it is not effective because you only use one column type. For example, you use 'age' column only in data for linear regression, so when I tried to click drop-down list, which I wonder if there is another column type, nothing happened. If you want to visualize data separately between data for linear regression and logistic regression, I think better to display normal graph with detail axis name. Or, you can combine them using @interact.
- In application part, please use the same plot between linear regression and logistic regression with detail axis name and scale. Also, for the accuracy part, please give the same unit, so we can compare it.
- In conclusion, it would be better if you can provide comparison table between linear regression and logistic regression, so it become more simple and attractive to read rather than line of text. 

Thank you.

**Comment by nikitml** [comment]
@hamzakpt @sobir-git please review
@ZamiraKholmatova please check comments, there's "action required"

**Review by hamzakpt — APPROVED** [review]

**Review by sobir-git — CHANGES_REQUESTED** [review]
And here a few notes.

I don't understand the purpose of the variable x (or slider x) in the two interactives. I think it's better to remove it because it just an unnecessary constraint for the amount of data to be plotted in the histogram.

A small fix:
- you have a piece of text `"(x,y)\in D"`. Make it `"$(x,y)\in D$"`  as it is latex code.

Logistic Regression:
- In the Logistic regression example, it would be good to tell readers what dataset are you going to work with. It's not obvious to guess I guess.
- The logistic regression example in the notebook shows a straight line in the output instead of a curved line. This image shows the difference, please take a look: 
image
- The explanation of the logistic regression example is a bit off. For example the relationship is stated differently than what is in Wikipedia. Also it says "where  b0  and  b1  represent the intercept and the slope of the linear regression considered above." while, if I remember correctly, there's no b0 and b1 in the case of logistic regression.

**Comment by sobir-git** [comment]
Good graph for Logistic regression plot !

No need for x-slider. Please remove it. Also rename the function to `show_data_distribution`. Overall like this:
```python
@interact
def show_data_distribution(column=df_logreg.columns):
    return df_logreg[column].hist()
```

**Fix**
You have said this in the logistic regression model:
> In the chosen dataset, the response falls into one of two classes (1 or 3).

But the code says differently:
`y = (iris["target"]==2).astype(np.int)` which means either 3rd class or the other two (1 and 2)

Either we have to edit the code or the text :)

**Comment by sobir-git** [comment]
One last fix:
The sentence should be: `In the chosen dataset, the response falls into one of two classes (Iris-Virginica or not Iris-Virginica). ` 

Because the classes `1 and 2` both mean non Iris-Virginica

**Comment by ZamiraKholmatova** [comment]
> One last fix:
> The sentence should be: `In the chosen dataset, the response falls into one of two classes (Iris-Virginica or not Iris-Virginica). `
> 
> Because the classes `1 and 2` both mean non Iris-Virginica

Fixed! Thanks!

**Review by artemkurylev — APPROVED** [review]

**Review by zakariabeni — APPROVED** [review]

**Review by sobir-git — APPROVED** [review]
